### ISSUE TITLE
Why does cAdvisor need so many host resources?

### ISSUE THREAD

**Issue by lonix1** [google/cadvisor#3453]
I'd like to use cAdvisor, which is set up like so:
```
sudo docker run \
  --volume=/:/rootfs:ro \                              😲
  --volume=/var/run:/var/run:ro \                      😲
  --volume=/sys:/sys:ro \                              😲
  --volume=/var/lib/docker/:/var/lib/docker:ro \       😲
  --volume=/dev/disk/:/dev/disk:ro \                   😲
  --publish=8080:8080 \
  --detach=true \
  --name=cadvisor \
  --privileged \                                       😲 😲 😲
  --device=/dev/kmsg \                                 😲
  gcr.io/cadvisor/cadvisor:v0.36.0
```

I've never had to expose so much of a host; and this runs in privileged mode too (and host sudo).

I know this is a reputable product, but before blindly copy-pasting this security nightmare, I'd like to know what it's doing and why are all those host resources required.

I'd typically expect the *docker socket* to be needed in such apps, yet here I'm basically exposing the entire host to the container... why? Are there perhaps docs or resources which discuss this?

(Also, maybe there's a less permissive config that I can use, without exposing my entire host to the container? Perhaps that config is meant as a "quickstart" for demos, but is not meant for a production deployment?)

Any advice or pointers would be appreciated, thanks!

**Comment by martadinata666** [comment]
because the system monitors, how will it monitor something with no resources to look up?

**Comment by martadinata666** [comment]
yes, it will monitor containers if you have one, but Cadvisor does not only monitor containers it does much more. If you only need to make cadvisor to monitor docker socket then just bind docker socket.

**Comment by martadinata666** [comment]
That is per person deployment, for starters remove a bit part then deploy, and look around the statistics. Keep reducing until you get what you want.

**Comment by SaturnusDJ** [comment]
I replaced `-v /:/rootfs/:ro` with `-v /var/lib/docker/:/var/lib/docker/:ro`

That full root access is really not cool.
